forge-core/src/main/java/forge/card/CardRulesPredicates.java

```diff
@@ -160,19 +145,16 @@ public static Predicate<CardRules> joinedType(final PredicateString.StringOp op,
     }
 
     public static Predicate<CardRules> hasCreatureType(final String... creatureTypes) {
-        return new Predicate<CardRules>() {
-            @Override
-            public boolean apply(final CardRules card) {
-                if (!card.getType().isCreature()) { return false; }
-
-                final Set<String> set = card.getType().getCreatureTypes();
-                for (final String creatureType : creatureTypes) {
-                    if (set.contains(creatureType)) {
-                        return true;
-                    }
+        return card -> {
+            if (!card.getType().isCreature()) { return false; }
+
+            final Set<String> set = card.getType().getCreatureTypes();
+            for (final String creatureType : creatureTypes) {
```

Comment:
might use `!Collections.disjoint` to check for overlapping

Reply:
see there: https://www.baeldung.com/java-check-elements-between-lists

Collections.disjoint returns true if there is no element from set A matching elements from set B,
so we just negate them.

it should stop and return false if first match is found (and then negated)

Reply:
Ah, I see what you mean. Done.